ui/components/NodeInputEmail.tsx

```diff
@@ -0,0 +1,30 @@
+import { getNodeLabel } from "@ory/integrations/ui";
+import { Input } from "@canonical/react-components";
+import React, { FC } from "react";
+import { NodeInputProps } from "./helpers";
+
+export const NodeInputEmail: FC<NodeInputProps> = ({
+  node,
+  setValue,
+  disabled,
+  dispatchSubmit,
+  error,
+}) => {
+  return (
+    <Input
+      type="email"
+      label={getNodeLabel(node)}
+      disabled={disabled}
+      defaultValue={node.messages.map(({ text }) => text).join(" ")}
+      error={error}
+      onChange={(e) => void setValue(e.target.value)}
+      onKeyDown={(e) => {
+        if (e.key === "Enter") {
+          e.preventDefault();
+          e.stopPropagation();
+          void dispatchSubmit(e, "code");
+        }
+      }}
```

Comment:
Would it make sense to add a check here for validating that it is an email?

Currently if the input is not an email, I get this
![image](https://github.com/canonical/identity-platform-login-ui/assets/19745916/f91deebd-063b-4879-b3ca-45d07665937c)

Kratos expects an email and because the input is not a valid email it returns an error that is not handled by the backend (the backend should handle this error when the error handling story is implemented)

Reply:
added frontend validation for the email input